pkg/util/version_utils.go

```diff
@@ -49,10 +55,23 @@ func (v Version) GreaterThan(other Version) bool {
 	return true
 }
 
+// SmallerThan determines whether this version is smaller than the given version
 func (v Version) SmallerThan(other Version) bool {
 	return other.GreaterThan(v)
 }
 
+// Valid returns whether this version is valid or not.
+// A version is considered to be "invalid" if it is equal to "0.0.0" or
+// contains a negative number, e.e. "0.-1.2"
+func (v Version) Valid() bool {
```

Comment:
Shouldn't this be InValid? 
According to your tests, Version{-1, 1, 1} would be valid

Reply:
fixedup - never happend :D